Fix#1889 (p:calendar timeControlType="select" does not work with national time pattern)

FastainCS09 opened the pull request:
The **if (hasTime()) Condition** in CalenderRenderer.java is buggy,it returns true only when the pattern is not null or contains (":") ,so that's why no pattern runs except that contains separator **(:)**.

Here is the code for checking time.
**public boolean hasTime() { 
     String pattern = getPattern(); 
     return (pattern != null && pattern.indexOf(":") != -1); 
}**
The work around is to comment the check and it will works for all the patterns alongwith any separator.
It has no consequence,even the user doesn't mention time with the pattern.

FastainCS09 committed:
Fix#1889

tandraschko commented:
So there is no reason that we had the if statement around the code block?

FastainCS09 commented:
Yeah,I checked it,even though , you donot mention time in the pattern,it will have no issue .

tandraschko commented:
| you donot mention time in the pattern,

not sure if in understand. It's possible to set time in the pattern.

FastainCS09 commented:
Just a continuation of the above comment, if u look at the funciton **hasTime()**
The last statement states : 
`return (pattern != null && pattern.indexOf(":") != -1);`

so if the pattern is "**MM:DD:YYYY**" ,,it still returns true,though the time is not given. So in any case,either  we comment the if block or check c only that the pattern is not null ,or other work aorund is to check,whether it has Pattern HH,MM,SS and then only returns true.

tandraschko commented:
Yeah but it's possible to set e.g. pattern="MM/dd/yyyy HH:mm”.

FastainCS09 commented:
yeah sure,i agree with you that we can add the time ,but if the pattern is e.g norwegian **MM.dd.yyyy HH.mm** then in that case ,it returns false.That was the problem,for which i propose that solution.

tandraschko commented:
Yeah but your fix looks just like an dirty workaround. We should fix the hasTime() method.

FastainCS09 commented:
If you want to retain the if code block ,then one solution  should be like this ,it is just a sample solution.

`return (pattern != null && pattern.contains("HH")  || pattern.contains("MM) || pattern.contains("SS");`

FastainCS09 commented:
if u want to improve the above code ,then feel free to give me your suggestion and i will then create a another updated pull request.

tandraschko commented:
yeah, something like that looks better.
but "MM" is month, we need to check "mm". "ss" are also lower case.

FastainCS09 commented:
Ok No issue,I will take care of lower case as well.So i will now create an updated Pull Request.Thanks for taking out time and giving me your feedback.

FastainCS09 committed:
Implemented time check functionality

Change @function hasTime() statement,by checking whether time format is present in the pattern and return true

FastainCS09 committed:
Fix Time check functionality

tandraschko commented:
Thanks!

FastainCS09 commented:
You welcome
